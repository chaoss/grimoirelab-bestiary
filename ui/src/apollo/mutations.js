import gql from "graphql-tag";

const ADD_PROJECT = gql`
  mutation addProject(
    $ecosystemId: ID
    $name: String
    $title: String
    $parentId: ID
  ) {
    addProject(
      ecosystemId: $ecosystemId
      name: $name
      title: $title
      parentId: $parentId
    ) {
      project {
        id
        name
      }
    }
  }
`;

const DELETE_PROJECT = gql`
  mutation deleteProject($id: ID) {
    deleteProject(id: $id) {
      project {
        id
      }
    }
  }
`;

const MOVE_PROJECT = gql`
  mutation moveProject($fromProjectId: ID, $toProjectId: ID) {
    moveProject(fromProjectId: $fromProjectId, toProjectId: $toProjectId) {
      project {
        id
        name
      }
    }
  }
`;

const UPDATE_PROJECT = gql`
  mutation updateProject($data: ProjectInputType, $id: ID) {
    updateProject(data: $data, id: $id) {
      project {
        id
        name
      }
    }
  }
`;

const ADD_ECOSYSTEM = gql`
  mutation addEcosystem($name: String!, $title: String, $description: String) {
    addEcosystem(name: $name, title: $title, description: $description) {
      ecosystem {
        id
      }
    }
  }
`;

const UPDATE_ECOSYSTEM = gql`
  mutation updateEcosystem($data: EcosystemInputType, $id: ID) {
    updateEcosystem(data: $data, id: $id) {
      ecosystem {
        id
      }
    }
  }
`;

const addProject = (apollo, data) => {
  const response = apollo.mutate({
    mutation: ADD_PROJECT,
    variables: {
      ecosystemId: data.ecosystemId,
      name: data.name,
      title: data.title,
      parentId: data.parentId
    }
  });
  return response;
};

const deleteProject = (apollo, id) => {
  const response = apollo.mutate({
    mutation: DELETE_PROJECT,
    variables: {
      id: id
    }
  });
  return response;
};

const moveProject = (apollo, fromProjectId, toProjectId) => {
  const response = apollo.mutate({
    mutation: MOVE_PROJECT,
    variables: {
      fromProjectId: fromProjectId,
      toProjectId: toProjectId
    }
  });
  return response;
};

const updateProject = (apollo, data, id) => {
  const response = apollo.mutate({
    mutation: UPDATE_PROJECT,
    variables: {
      data: data,
      id: id
    }
  });
  return response;
};

const addEcosystem = (apollo, data) => {
  const response = apollo.mutate({
    mutation: ADD_ECOSYSTEM,
    variables: {
      name: data.name,
      title: data.title,
      description: data.description
    }
  });
  return response;
};

const updateEcosystem = (apollo, data, id) => {
  const response = apollo.mutate({
    mutation: UPDATE_ECOSYSTEM,
    variables: {
      data: data,
      id: id
    }
  });
  return response;
};

export {
  addProject,
  deleteProject,
  moveProject,
  updateProject,
  addEcosystem,
  updateEcosystem
};
