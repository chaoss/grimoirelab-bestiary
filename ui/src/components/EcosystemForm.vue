<template>
  <v-form ref="form">
    <v-row>
      <v-col cols="4">
        <v-text-field
          v-model="form.title"
          :rules="validations.required"
          label="Ecosystem title"
          outlined
          dense
          @change="suggestName"
        ></v-text-field>
      </v-col>
    </v-row>
    <v-row>
      <v-col cols="4">
        <v-text-field
          v-model="form.name"
          :rules="validations.required"
          label="Ecosystem name"
          outlined
          dense
          @input="touchedName = true"
        ></v-text-field>
      </v-col>
    </v-row>
    <v-row>
      <v-col cols="4">
        <v-textarea
          v-model="form.description"
          label="Ecosystem description (optional)"
          :rules="validations.maxLength"
          outlined
          dense
          counter="128"
          rows="4"
        ></v-textarea>
      </v-col>
    </v-row>
    <v-row>
      <v-col cols="4" class="d-flex justify-end">
        <v-btn color="primary" depressed @click="save">
          Save
        </v-btn>
      </v-col>
    </v-row>
    <v-row>
      <v-col cols="4">
        <v-alert v-if="error" text outlined type="error">
          {{ error }}
        </v-alert>
      </v-col>
    </v-row>
  </v-form>
</template>

<script>
export default {
  name: "EcosystemForm",
  props: {
    saveFunction: {
      type: Function,
      required: true
    },
    id: {
      type: [Number, String],
      required: false
    },
    name: {
      type: String,
      required: false
    },
    title: {
      type: String,
      required: false
    },
    description: {
      type: String,
      required: false
    }
  },
  data() {
    return {
      form: {
        name: this.name,
        title: this.title,
        description: this.description
      },
      validations: {
        required: [v => !!v || "Required"],
        maxLength: [v => (v ? v.length <= 128 : true) || "Max 128 characters"]
      },
      touchedName: false,
      error: null
    };
  },
  methods: {
    suggestName(value) {
      if (value && !this.name && !this.touchedName) {
        this.form.name = value
          .trim()
          .replace(/\s+/g, "-")
          .toLowerCase();
      }
    },
    async save() {
      if (!this.$refs.form.validate()) {
        return;
      }
      try {
        const data = {
          name: this.form.name.trim(),
          title: this.form.title.trim(),
          description: this.form.description
        };
        const response = await this.saveFunction(data);
        if (response && !response.errors) {
          this.$router.push({ path: `/ecosystem/${response.id}` });
        } else if (response.errors) {
          this.error = response.errors[0].message;
        }
      } catch (error) {
        this.error = error;
      }
    }
  }
};
</script>
