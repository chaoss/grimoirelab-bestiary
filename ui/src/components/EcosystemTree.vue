<template>
  <div>
    <v-treeview
      v-if="items"
      dense
      hoverable
      open-all
      expand-icon="mdi-chevron-down"
      active-class="dropzone"
      item-key="name"
      item-text="title"
      item-children="subprojects"
      :items="items"
      :active="active"
    >
      <template v-slot:label="{ item }">
        <div
          class="d-flex justify-space-between align-center hidden"
          :draggable="!item.projectSet"
          @dragstart.stop="startDrag(item, $event)"
          @drop.stop.prevent="onDrop(item, $event)"
          @dragenter.prevent
          @dragover.prevent="handleDrag(item, $event, true)"
          @dragleave.prevent="handleDrag(item, $event, false)"
          @dragend.prevent="onDragend"
        >
          <router-link
            v-slot="{ navigate, href, isExactActive }"
            :to="getLink(item)"
            custom
          >
            <a
              class="router-link"
              :class="{ 'router-link-exact-active': isExactActive }"
              :href="href"
              @click="navigate"
              @drop.prevent="onDrop(item, $event)"
            >
              {{ item.title }}
            </a>
          </router-link>

          <v-menu offset-y v-if="!item.projectSet">
            <template v-slot:activator="{ on }">
              <v-btn v-on="on" icon color="transparent">
                <v-icon>mdi-dots-horizontal</v-icon>
              </v-btn>
            </template>
            <v-list dense>
              <v-list-item
                :to="{
                  name: 'project-edit',
                  params: {
                    id: item.ecosystem.id,
                    name: item.name
                  }
                }"
              >
                <v-list-item-icon class="mr-2">
                  <v-icon small color="#3f3f3f">mdi-pencil-outline</v-icon>
                </v-list-item-icon>
                <v-list-item-title>Edit</v-list-item-title>
              </v-list-item>
              <v-list-item @click="confirmDeleteProject(item)">
                <v-list-item-icon class="mr-2">
                  <v-icon small color="#3f3f3f">mdi-trash-can-outline</v-icon>
                </v-list-item-icon>
                <v-list-item-title>Delete</v-list-item-title>
              </v-list-item>
            </v-list>
          </v-menu>

          <v-menu offset-y v-else>
            <template v-slot:activator="{ on }">
              <v-btn v-on="on" icon color="transparent">
                <v-icon>mdi-dots-horizontal</v-icon>
              </v-btn>
            </template>
            <v-list dense>
              <v-list-item
                :to="{ name: 'project-new', params: { id: item.id } }"
              >
                <v-list-item-icon class="mr-2">
                  <v-icon small color="#3f3f3f">mdi-plus</v-icon>
                </v-list-item-icon>
                <v-list-item-title>Add a project</v-list-item-title>
              </v-list-item>
              <v-list-item
                :to="{ name: 'ecosystem-edit', params: { id: item.id } }"
              >
                <v-list-item-icon class="mr-2">
                  <v-icon small color="#3f3f3f">mdi-pencil-outline</v-icon>
                </v-list-item-icon>
                <v-list-item-title>Edit</v-list-item-title>
              </v-list-item>
              <v-list-item @click="confirmDeleteEcosystem(item)">
                <v-list-item-icon class="mr-2">
                  <v-icon small color="#3f3f3f">mdi-trash-can-outline</v-icon>
                </v-list-item-icon>
                <v-list-item-title>Delete</v-list-item-title>
              </v-list-item>
            </v-list>
          </v-menu>
        </div>
      </template>
    </v-treeview>
    <div class="drag-image"></div>
  </div>
</template>

<script>
import { hasSameRoot, isDescendant, isParent } from "../utils/projects";
export default {
  name: "EcosystemTree",
  props: {
    ecosystem: {
      type: Object,
      required: true
    },
    deleteProject: {
      type: Function,
      required: true
    },
    moveProject: {
      type: Function,
      required: true
    },
    deleteEcosystem: {
      type: Function,
      required: true
    }
  },
  data() {
    return {
      items: null,
      dragged: null,
      active: []
    };
  },
  methods: {
    filterDuplicateProjects(ecosystem) {
      if (this.ecosystem.projectSet) {
        const subprojects = this.ecosystem.projectSet.filter(
          project => !project.parentProject
        );
        return [Object.assign(ecosystem, { subprojects })];
      } else {
        return [ecosystem];
      }
    },
    getLink(item) {
      if (item.projectSet) {
        return `/ecosystem/${item.id}`;
      } else {
        return `/ecosystem/${item.ecosystem.id}/project/${item.name}`;
      }
    },
    confirmDeleteProject(item) {
      const dialog = {
        isOpen: true,
        title: `Delete project ${item.title}?`,
        warning: "This will delete every project inside it.",
        action: () => this.deleteProject(item.id)
      };
      this.$store.commit("setDialog", dialog);
    },
    confirmDeleteEcosystem(item) {
      const dialog = {
        isOpen: true,
        title: `Delete ecosystem ${item.title}?`,
        warning: "This will delete every project inside it.",
        action: () => this.deleteEcosystem(item.id)
      };
      this.$store.commit("setDialog", dialog);
    },
    startDrag(item, event) {
      if (item.projectSet) {
        return;
      }
      const dragImage = document.querySelector(".drag-image");
      dragImage.innerHTML = event.target.closest(".v-treeview-node").innerHTML;
      event.dataTransfer.setDragImage(dragImage, 10, 10);
      event.dataTransfer.setData("move", item.id);
      this.dragged = item;
    },
    handleDrag(item, event, dragged) {
      if (dragged && this.allowDrag(item)) {
        this.active.push(item.name);
      } else {
        this.active = this.active.filter(project => project.name === item.name);
      }
    },
    allowDrag(item) {
      if (this.dragged && item.ecosystem) {
        return (
          item.ecosystem.id === this.dragged.ecosystem.id &&
          hasSameRoot(this.dragged, item) &&
          !isParent(this.dragged, item) &&
          !isDescendant(item, this.dragged) &&
          item.id !== this.dragged.id
        );
      }
      return false;
    },
    onDrop(item) {
      if (this.dragged && this.allowDrag(item)) {
        const project = this.dragged.id;
        const dialog = {
          isOpen: true,
          title: `Move project ${this.dragged.title} to ${item.title}?`,
          action: () => this.moveProject(project, item.id)
        };
        this.$store.commit("setDialog", dialog);
        this.active = this.active.filter(project => project.name === item.name);
        this.dragged = null;
      }
    },
    onDragend() {
      this.dragged = null;
      this.active = [];
    }
  },
  mounted() {
    this.items = this.filterDuplicateProjects(this.ecosystem);
  },
  watch: {
    ecosystem(value) {
      this.items = this.filterDuplicateProjects(value);
    }
  }
};
</script>

<style lang="scss" scoped>
@import "../styles/_variables";
.v-treeview-node__root a {
  font-family: "Roboto", sans-serif;
  font-size: 0.875rem;
  color: $text-color;
  letter-spacing: 0.2px;
}
::v-deep .v-treeview-node__children {
  .theme--light.v-icon,
  a {
    color: $text-color--light;
  }
}
.v-treeview-node__label a {
  text-decoration: none;
  font-weight: 500;
}
.v-treeview-node__root .router-link-exact-active {
  color: $primary-color;
  font-weight: 700;
}
::v-deep .mdi::before,
.mdi-set {
  font-size: 18px;
}
::v-deep .v-treeview-node__toggle,
::v-deep .v-treeview-node__level {
  width: 18px;
}
::v-deep .v-treeview-node--active {
  .theme--light.v-icon,
  .hidden:hover .v-icon.v-icon {
    color: $primary-color;
  }
}
.hidden {
  .v-icon.v-icon {
    color: transparent;
  }
  &:hover,
  &:focus,
  .v-btn:focus {
    .v-icon.v-icon {
      color: $text-color;
    }
    .router-link {
      overflow: hidden;
      text-overflow: ellipsis;
    }
  }
}
.v-btn--icon {
  min-width: 36px;
}
::v-deep .dropzone {
  &::before {
    border: thin solid;
    background: lighten($primary-color, 30%);
    opacity: 0.12;
    transition: none;
  }
}
.drag-image {
  position: absolute;
  top: -1000px;
  left: -1000px;
  background-color: #ffffff;
  border: 2px solid #003756;

  .v-treeview-node__level:first-child {
    display: none;
  }
}
</style>
