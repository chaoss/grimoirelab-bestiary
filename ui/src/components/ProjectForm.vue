<template>
  <v-form ref="form">
    <v-row>
      <v-col cols="4">
        <v-text-field
          v-model="title"
          label="Project title"
          :rules="validations.required"
          outlined
          dense
          @change="suggestName"
        ></v-text-field>
      </v-col>
    </v-row>
    <v-row>
      <v-col cols="4">
        <v-text-field
          v-model="name"
          label="Project name"
          :rules="validations.required"
          outlined
          dense
          @input="touchedName = true"
        ></v-text-field>
      </v-col>
    </v-row>
    <v-row>
      <v-col cols="4">
        <v-autocomplete
          v-model="parentId"
          :items="projects"
          label="Parent project (optional)"
          item-text="title"
          item-value="id"
          clearable
          outlined
          dense
          @click.once="loadParentProjects"
        >
          <template v-slot:no-data>
            <v-list-item>
              <v-list-item-content>
                <v-list-item-title class="text--disabled">
                  No available projects
                </v-list-item-title>
              </v-list-item-content>
            </v-list-item>
          </template>
        </v-autocomplete>
      </v-col>
    </v-row>
    <v-row>
      <v-col cols="4" class="d-flex justify-end">
        <v-btn color="primary" depressed @click="save">
          Save
        </v-btn>
      </v-col>
    </v-row>
  </v-form>
</template>

<script>
export default {
  name: "ProjectForm",
  props: {
    ecosystemId: {
      type: Number,
      required: false
    },
    getProjects: {
      type: Function,
      required: true
    },
    addProject: {
      type: Function,
      required: true
    },
    parent: {
      type: Object,
      required: false,
      default: null
    }
  },
  data() {
    return {
      name: null,
      title: null,
      parentId: null,
      touchedName: false,
      projects: [],
      validations: {
        required: [value => !!value || "Required"]
      }
    };
  },
  methods: {
    suggestName(value) {
      if (value && !this.touchedName) {
        this.name = value
          .trim()
          .replace(/\s+/g, "-")
          .toLowerCase();
      }
    },
    async loadParentProjects() {
      const response = await this.getProjects(Number(this.ecosystemId));
      if (response) {
        this.projects = response;
      }
    },
    async save() {
      if (!this.$refs.form.validate()) {
        return;
      }
      const data = {
        name: this.name.trim(),
        title: this.title,
        parentId: this.parentId ? Number(this.parentId) : null,
        ecosystemId: this.ecosystemId
      };
      const response = await this.addProject(data);
      if (response) {
        const projectName = response.name;
        this.$router.push({
          path: `/ecosystem/${this.ecosystemId}/project/${projectName}`
        });
      }
    }
  },
  created() {
    if (this.parent) {
      this.loadParentProjects();
      this.parentId = this.parent.id;
    }
  }
};
</script>
