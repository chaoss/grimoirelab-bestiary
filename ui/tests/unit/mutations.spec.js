import { shallowMount } from "@vue/test-utils";
import Vue from "vue";
import Vuetify from "vuetify";
import * as Mutations from "@/apollo/mutations";
import ProjectForm from "@/components/ProjectForm";
import EcosystemForm from "@/components/EcosystemForm";
import EcosystemTree from "@/components/EcosystemTree";

Vue.use(Vuetify);

describe("ProjectsForm mutations", () => {
  const mountFunction = options => {
    return shallowMount(ProjectForm, {
      Vue,
      Vuetify,
      propsData: {
        ecosystemId: 1,
        getProjects: () => {},
        saveFunction: () => {}
      },
      ...options
    });
  };

  const addProjectResponse = {
    data: {
      addProject: {
        project: {
          id: "1",
          name: "new-project",
          __typename: "ProjectType"
        },
        __typename: "AddProject"
      }
    }
  };

  const updateProjectResponse = {
    data: {
      updateProject: {
        project: {
          id: "39",
          name: "test",
          __typename: "ProjectType"
        },
        __typename: "UpdateProject"
      }
    }
  };

  test("Mock mutation for addProject", async () => {
    const mutate = jest.fn(() => Promise.resolve(addProjectResponse));
    const wrapper = mountFunction({
      mocks: {
        $apollo: {
          mutate
        }
      },
      data() {
        return {
          form: {
            name: "New Project",
            title: "new-project"
          }
        };
      }
    });
    await wrapper.setProps({ saveFunction: mutate });
    await Mutations.addProject(wrapper.vm.$apollo, {
      name: wrapper.vm.form.name,
      title: wrapper.vm.form.title,
      ecosystemId: wrapper.vm.ecosystemId
    });

    expect(mutate).toBeCalled();
    expect(wrapper.element).toMatchSnapshot();
  });

  test("Mock mutation for updateProject", async () => {
    const mutate = jest.fn(() => Promise.resolve(updateProjectResponse));
    const wrapper = mountFunction({
      mocks: {
        $apollo: {
          mutate
        },
        $router: {
          push: () => {}
        }
      },
      data() {
        return {
          form: {
            name: "test",
            title: "Test Title",
            parentId: 2
          }
        };
      }
    });
    await wrapper.setProps({ saveFunction: mutate });

    //Mock Vuetify form validation
    wrapper.vm.$refs.form.validate = () => {
      return true;
    };

    await wrapper.vm.save();

    expect(mutate).toBeCalledWith({
      ecosystemId: 1,
      name: "test",
      parentId: 2,
      title: "Test Title"
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});

describe("Ecosystem mutations", () => {
  const response = {
    data: {
      addEcosystem: {
        ecosystem: {
          id: "1"
        }
      }
    }
  };

  test("Mock mutation for addEcosystem", async () => {
    const mutate = jest.fn(() => Promise.resolve(response));
    const wrapper = shallowMount(EcosystemForm, {
       Vue,
       mocks: {
         $apollo: {
           mutate
         }
       },
       propsData: {
         saveFunction: mutate
       },
       data() {
        return {
          form: {
            name: "test",
            title: "Test",
            description: "Lorem ipsum"
          }
        };
       }
    });

    await Mutations.addEcosystem(wrapper.vm.$apollo, {
      name: wrapper.vm.form.name,
      title: wrapper.vm.form.title,
      description: wrapper.vm.form.description
    });

    expect(mutate).toBeCalled();
    expect(wrapper.element).toMatchSnapshot();
  });

  test("Mock mutation for updateEcosystem", async () => {
    const mutate = jest.fn(() => Promise.resolve(response));
    const wrapper = shallowMount(EcosystemForm, {
       Vue,
       mocks: {
         $apollo: {
           mutate
         }
       },
       propsData: {
         saveFunction: mutate,
         name: "test",
         title: "Test",
         description: "Lorem ipsum"
       }
    });

    await Mutations.updateEcosystem(wrapper.vm.$apollo, {
      name: wrapper.vm.form.name,
      title: wrapper.vm.form.title,
      description: wrapper.vm.form.description
    });

    expect(mutate).toBeCalled();
    expect(wrapper.element).toMatchSnapshot();
  });

  test("Mock mutation for deleteEcosystem", async () => {
    const mutate = jest.fn(() => Promise.resolve(response));
    const wrapper = shallowMount(EcosystemTree, {
       Vue,
       mocks: {
         $apollo: {
           mutate
         }
       },
       propsData: {
         ecosystem: {
           id: 1,
           title: "Test"
         },
         deleteEcosystem: mutate,
         deleteProject: () => {},
         moveProject: () => {}
       }
    });

    await Mutations.deleteEcosystem(
      wrapper.vm.$apollo,
      wrapper.vm.ecosystem.id
    );

    expect(mutate).toBeCalled();
    expect(wrapper.element).toMatchSnapshot();
  });
});
