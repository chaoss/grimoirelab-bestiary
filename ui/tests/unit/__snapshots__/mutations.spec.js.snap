// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ecosystem mutations Mock mutation for addEcosystem 1`] = `
<v-form-stub>
  <v-row-stub
    tag="div"
  >
    <v-col-stub
      cols="4"
      tag="div"
    >
      <v-text-field-stub
        backgroundcolor=""
        clearicon="$clear"
        dense="true"
        errorcount="1"
        errormessages=""
        label="Ecosystem title"
        loaderheight="2"
        messages=""
        outlined="true"
        rules="v => !!v || \\"Required\\""
        successmessages=""
        type="text"
        value="Test"
      />
    </v-col-stub>
  </v-row-stub>
   
  <v-row-stub
    tag="div"
  >
    <v-col-stub
      cols="4"
      tag="div"
    >
      <v-text-field-stub
        backgroundcolor=""
        clearicon="$clear"
        dense="true"
        errorcount="1"
        errormessages=""
        label="Ecosystem name"
        loaderheight="2"
        messages=""
        outlined="true"
        rules="v => !!v || \\"Required\\""
        successmessages=""
        type="text"
        value="test"
      />
    </v-col-stub>
  </v-row-stub>
   
  <v-row-stub
    tag="div"
  >
    <v-col-stub
      cols="4"
      tag="div"
    >
      <v-textarea-stub
        backgroundcolor=""
        clearicon="$clear"
        counter="128"
        dense="true"
        errorcount="1"
        errormessages=""
        label="Ecosystem description (optional)"
        loaderheight="2"
        messages=""
        outlined="true"
        rowheight="24"
        rows="4"
        rules="v => (v ? v.length <= 128 : true) || \\"Max 128 characters\\""
        successmessages=""
        type="text"
        value="Lorem ipsum"
      />
    </v-col-stub>
  </v-row-stub>
   
  <v-row-stub
    tag="div"
  >
    <v-col-stub
      class="d-flex justify-end"
      cols="4"
      tag="div"
    >
      <v-btn-stub
        activeclass=""
        color="primary"
        depressed="true"
        tag="button"
        type="button"
      >
        
        Save
      
      </v-btn-stub>
    </v-col-stub>
  </v-row-stub>
   
  <v-row-stub
    tag="div"
  >
    <v-col-stub
      cols="4"
      tag="div"
    >
      <!---->
    </v-col-stub>
  </v-row-stub>
</v-form-stub>
`;

exports[`Ecosystem mutations Mock mutation for deleteEcosystem 1`] = `
<div>
  <v-treeview-stub
    active=""
    activeclass="dropzone"
    color="primary"
    dense="true"
    expandicon="mdi-chevron-down"
    hoverable="true"
    indeterminateicon="$checkboxIndeterminate"
    itemchildren="subprojects"
    itemdisabled="disabled"
    itemkey="name"
    items="[object Object]"
    itemtext="title"
    loadingicon="$loading"
    officon="$checkboxOff"
    onicon="$checkboxOn"
    open=""
    openall="true"
    selectedcolor="accent"
    selectiontype="leaf"
    value=""
  />
   
  <div
    class="drag-image"
  />
</div>
`;

exports[`Ecosystem mutations Mock mutation for updateEcosystem 1`] = `
<v-form-stub>
  <v-row-stub
    tag="div"
  >
    <v-col-stub
      cols="4"
      tag="div"
    >
      <v-text-field-stub
        backgroundcolor=""
        clearicon="$clear"
        dense="true"
        errorcount="1"
        errormessages=""
        label="Ecosystem title"
        loaderheight="2"
        messages=""
        outlined="true"
        rules="v => !!v || \\"Required\\""
        successmessages=""
        type="text"
        value="Test"
      />
    </v-col-stub>
  </v-row-stub>
   
  <v-row-stub
    tag="div"
  >
    <v-col-stub
      cols="4"
      tag="div"
    >
      <v-text-field-stub
        backgroundcolor=""
        clearicon="$clear"
        dense="true"
        errorcount="1"
        errormessages=""
        label="Ecosystem name"
        loaderheight="2"
        messages=""
        outlined="true"
        rules="v => !!v || \\"Required\\""
        successmessages=""
        type="text"
        value="test"
      />
    </v-col-stub>
  </v-row-stub>
   
  <v-row-stub
    tag="div"
  >
    <v-col-stub
      cols="4"
      tag="div"
    >
      <v-textarea-stub
        backgroundcolor=""
        clearicon="$clear"
        counter="128"
        dense="true"
        errorcount="1"
        errormessages=""
        label="Ecosystem description (optional)"
        loaderheight="2"
        messages=""
        outlined="true"
        rowheight="24"
        rows="4"
        rules="v => (v ? v.length <= 128 : true) || \\"Max 128 characters\\""
        successmessages=""
        type="text"
        value="Lorem ipsum"
      />
    </v-col-stub>
  </v-row-stub>
   
  <v-row-stub
    tag="div"
  >
    <v-col-stub
      class="d-flex justify-end"
      cols="4"
      tag="div"
    >
      <v-btn-stub
        activeclass=""
        color="primary"
        depressed="true"
        tag="button"
        type="button"
      >
        
        Save
      
      </v-btn-stub>
    </v-col-stub>
  </v-row-stub>
   
  <v-row-stub
    tag="div"
  >
    <v-col-stub
      cols="4"
      tag="div"
    >
      <!---->
    </v-col-stub>
  </v-row-stub>
</v-form-stub>
`;

exports[`ProjectsForm mutations Mock mutation for addProject 1`] = `
<v-form-stub>
  <v-row-stub
    tag="div"
  >
    <v-col-stub
      cols="4"
      tag="div"
    >
      <v-text-field-stub
        backgroundcolor=""
        clearicon="$clear"
        dense="true"
        errorcount="1"
        errormessages=""
        label="Project title"
        loaderheight="2"
        messages=""
        outlined="true"
        rules="value => !!value || \\"Required\\""
        successmessages=""
        type="text"
        value="new-project"
      />
    </v-col-stub>
  </v-row-stub>
   
  <v-row-stub
    tag="div"
  >
    <v-col-stub
      cols="4"
      tag="div"
    >
      <v-text-field-stub
        backgroundcolor=""
        clearicon="$clear"
        dense="true"
        errorcount="1"
        errormessages=""
        label="Project name"
        loaderheight="2"
        messages=""
        outlined="true"
        rules="value => !!value || \\"Required\\""
        successmessages=""
        type="text"
        value="New Project"
      />
    </v-col-stub>
  </v-row-stub>
   
  <v-row-stub
    tag="div"
  >
    <v-col-stub
      cols="4"
      tag="div"
    >
      <v-autocomplete-stub
        allowoverflow="true"
        appendicon="$dropdown"
        backgroundcolor=""
        clearable="true"
        clearicon="$clear"
        dense="true"
        errorcount="1"
        errormessages=""
        filter="function _default(item, queryText, itemText) {
        return itemText.toLocaleLowerCase().indexOf(queryText.toLocaleLowerCase()) > -1;
      }"
        itemcolor="primary"
        itemdisabled="disabled"
        items=""
        itemtext="title"
        itemvalue="id"
        label="Parent project (optional)"
        loaderheight="2"
        menuprops="[object Object]"
        messages=""
        nodatatext="$vuetify.noDataText"
        outlined="true"
        rules=""
        successmessages=""
        type="text"
        valuecomparator="function deepEqual(a, b) {
  if (a === b) return true;

  if (a instanceof Date && b instanceof Date && a.getTime() !== b.getTime()) {
    // If the values are Date, compare them as timestamps
    return false;
  }

  if (a !== Object(a) || b !== Object(b)) {
    // If the values aren't objects, they were already checked for equality
    return false;
  }

  var props = Object.keys(a);

  if (props.length !== Object.keys(b).length) {
    // Different number of props, don't bother to check
    return false;
  }

  return props.every(function (p) {
    return deepEqual(a[p], b[p]);
  });
}"
      />
    </v-col-stub>
  </v-row-stub>
   
  <v-row-stub
    tag="div"
  >
    <v-col-stub
      class="d-flex justify-end"
      cols="4"
      tag="div"
    >
      <v-btn-stub
        activeclass=""
        color="primary"
        depressed="true"
        tag="button"
        type="button"
      >
        
        Save
      
      </v-btn-stub>
    </v-col-stub>
  </v-row-stub>
</v-form-stub>
`;

exports[`ProjectsForm mutations Mock mutation for updateProject 1`] = `
<v-form-stub>
  <v-row-stub
    tag="div"
  >
    <v-col-stub
      cols="4"
      tag="div"
    >
      <v-text-field-stub
        backgroundcolor=""
        clearicon="$clear"
        dense="true"
        errorcount="1"
        errormessages=""
        label="Project title"
        loaderheight="2"
        messages=""
        outlined="true"
        rules="value => !!value || \\"Required\\""
        successmessages=""
        type="text"
        value="Test Title"
      />
    </v-col-stub>
  </v-row-stub>
   
  <v-row-stub
    tag="div"
  >
    <v-col-stub
      cols="4"
      tag="div"
    >
      <v-text-field-stub
        backgroundcolor=""
        clearicon="$clear"
        dense="true"
        errorcount="1"
        errormessages=""
        label="Project name"
        loaderheight="2"
        messages=""
        outlined="true"
        rules="value => !!value || \\"Required\\""
        successmessages=""
        type="text"
        value="test"
      />
    </v-col-stub>
  </v-row-stub>
   
  <v-row-stub
    tag="div"
  >
    <v-col-stub
      cols="4"
      tag="div"
    >
      <v-autocomplete-stub
        allowoverflow="true"
        appendicon="$dropdown"
        backgroundcolor=""
        clearable="true"
        clearicon="$clear"
        dense="true"
        errorcount="1"
        errormessages=""
        filter="function _default(item, queryText, itemText) {
        return itemText.toLocaleLowerCase().indexOf(queryText.toLocaleLowerCase()) > -1;
      }"
        itemcolor="primary"
        itemdisabled="disabled"
        items=""
        itemtext="title"
        itemvalue="id"
        label="Parent project (optional)"
        loaderheight="2"
        menuprops="[object Object]"
        messages=""
        nodatatext="$vuetify.noDataText"
        outlined="true"
        rules=""
        successmessages=""
        type="text"
        value="2"
        valuecomparator="function deepEqual(a, b) {
  if (a === b) return true;

  if (a instanceof Date && b instanceof Date && a.getTime() !== b.getTime()) {
    // If the values are Date, compare them as timestamps
    return false;
  }

  if (a !== Object(a) || b !== Object(b)) {
    // If the values aren't objects, they were already checked for equality
    return false;
  }

  var props = Object.keys(a);

  if (props.length !== Object.keys(b).length) {
    // Different number of props, don't bother to check
    return false;
  }

  return props.every(function (p) {
    return deepEqual(a[p], b[p]);
  });
}"
      />
    </v-col-stub>
  </v-row-stub>
   
  <v-row-stub
    tag="div"
  >
    <v-col-stub
      class="d-flex justify-end"
      cols="4"
      tag="div"
    >
      <v-btn-stub
        activeclass=""
        color="primary"
        depressed="true"
        tag="button"
        type="button"
      >
        
        Save
      
      </v-btn-stub>
    </v-col-stub>
  </v-row-stub>
</v-form-stub>
`;
